### DISCUSSION TITLE
Problem with pyhyp meshing

### DISCUSSION THREAD

**Discussion by FaiqShahbaz** [mdolab/dafoam#699] [General]
Initially i was facing the error below, which i guess occurred because the formatting of airfoil coordinates was not exact. Got rid of this error after making sure that both upper and lower side points are equal (which is weird since in tutorial S809 airfoil the no of points for both sides were'nt exact and yet it worked, so what could be possible reason for this error occuring.
dafoamuser@70d0812a9d6e:~/mount$ ./preProcessing.sh
/bin/bash: /home/dafoamuser/dafoam/packages/miniconda3/lib/libtinfo.so.6: no version information available (required by /bin/bash)
Generating mesh..
--> FOAM FATAL IO ERROR:
Attempt to get back from bad stream
file: [volumeMesh.xyz](http://volumemesh.xyz/) at line 1.
    From function bool Foam::Istream::getBack(Foam::token&)
    in file db/IOstreams/IOstreams/Istream.C at line 56.
FOAM exiting

and now the python file for generating mesh is running but marchDist is not being uniform and mesh i being distorted for some reason. i have attached image and logmesh file, i will be really gald if someone could help me with this. The airfoil i am using is SG6043.

![image_2024-10-28_071609749](https://github.com/user-attachments/assets/39a4e567-173e-4183-a38f-68b41023c15a)

[logMeshGeneration.txt](https://github.com/user-attachments/files/17541606/logMeshGeneration.txt)

**Comment by friedenhe** [comment]
The upper and lower airfoil profiles do not need to have the same number of points, but you need to manually truncate the last few points near the trailing edge. Also, both upper and lower profiles need to go from the leading to the trailing edge.

**Reply by FaiqShahbaz** [reply]
I initially did the same but there is still problem with the meshing, i get the same mesh as i have attached before, i have also attached the airfoil points and image so you can check if i am using it correctly or not. I think the error is arising in mesh phase since its making two different overlapping domains.
![380695241-39a4e567-173e-4183-a38f-68b41023c15a](https://github.com/user-attachments/assets/e3d3bd5f-35d9-4503-9feb-e88c04d773b4)

![ce2b869d-51d0-4b48-ba79-0447d9cd8962](https://github.com/user-attachments/assets/35b1d958-a082-49c0-b75a-d9dfb31f11d0)
PS Profile                                                
0.000000 0.000000
0.001496 0.007410
0.000024 0.000942
0.000588 -.004677
0.003930 -.008595
0.010772 -.011195
0.020868 -.013164
0.034086 -.014302
0.050504 -.014598
0.070145 -.014215
0.092915 -.013293
0.118637 -.011970
0.147117 -.010343
0.178147 -.008477
0.211498 -.006446
0.246927 -.004303
0.284172 -.002104
0.322960 0.000114
0.363008 0.002319
0.404014 0.004498
0.445705 0.006689
0.487849 0.008933
0.530290 0.011167
0.572862 0.013270
0.615252 0.015037
0.657041 0.016351
0.697849 0.017212
0.737347 0.017604
0.775201 0.017514
0.811085 0.016945
0.844678 0.015916
0.875671 0.014467
0.903773 0.012654
0.928712 0.010558
0.950241 0.008276
0.968141 0.005908

SS Profile
  0.000000 0.0000000
  0.001496  0.007410
  0.005277  0.014355
  0.011416  0.021706
  0.019828  0.029159
  0.030622  0.036590
  0.043757  0.043981
  0.059198  0.051170
  0.076936  0.058094
  0.096866  0.064730
  0.118895  0.070995
  0.142926  0.076839
  0.168817  0.082199
  0.196447  0.087021
  0.225671  0.091274
  0.256330  0.094905
  0.288269  0.097880
  0.321306  0.100174
  0.355272  0.101752
  0.389995  0.102599
  0.425276  0.102690
  0.460955  0.102010
  0.496847  0.100592
  0.532761  0.098415
  0.568537  0.095505
  0.603962  0.091903
  0.638892  0.087606
  0.673195  0.082684
  0.706663  0.077173
  0.739215  0.071138
  0.770741  0.064777
  0.801008  0.058240
  0.829794  0.051648
  0.856884  0.045099
  0.882072  0.038683
  0.905161  0.032475
  0.925966  0.026547
  0.944288  0.020916
  0.960071  0.015523
  0.973434  0.010465

**Comment by friedenhe** [comment]
Your profiles have problems. The PS profile's x is not monotonic. If you remove the 3rd point, it should work

>
PS Profile
0.000000 0.000000
0.001496 0.007410
0.000024 0.000942
0.000588 -.004677
